cluster-operator/src/main/java/io/strimzi/operator/cluster/model/cruisecontrol/DiskCapacity.java

```diff
@@ -4,49 +4,132 @@
+import io.strimzi.operator.cluster.model.VolumeUtils;
 import io.vertx.core.json.JsonObject;
 
 import java.util.HashMap;
 import java.util.Map;
 
 /**
- * Cruise Control disk capacity
+ * Cruise Control disk capacity configuration for broker.
  */
 public class DiskCapacity {
+    private static final String DEFAULT_DISK_CAPACITY_IN_MIB = "100000";
+    private static final String KAFKA_MOUNT_PATH = "/var/lib/kafka";
+    private static final String KAFKA_LOG_DIR = "kafka-log";
     private static final String SINGLE_DISK = "";
-    private Map<String, String> map;
+
+    private final Map<String, String> config;
 
     /**
      * Constructor
+     *
+     * Generate JBOD disk capacity configuration for a broker using the supplied storage configuration.
+     *
+     * @param storage Storage configuration for Kafka cluster.
+     * @param brokerId Id of the broker.
      */
-    public DiskCapacity() {
-        map = new HashMap<>(1);
+    protected DiskCapacity(Storage storage, int brokerId) {
+        if (storage instanceof JbodStorage jbodStorage) {
+            config = generateJbodDiskConfig(jbodStorage, brokerId);
+        } else {
+            config = generateNonJbodDiskConfig(storage);
+        }
     }
 
-    private DiskCapacity(String size) {
-        this();
-        map.put(SINGLE_DISK, size);
+    /**
+     * Parse a K8S-style representation of a disk size, such as {@code 100Gi},
+     * into the equivalent number of mebibytes represented as a String.
+     *
+     * @param size The String representation of the volume size.
+     * @return The equivalent number of mebibytes.
+     */
+    private static String getSizeInMiB(String size) {
+        if (size == null) {
+            return DEFAULT_DISK_CAPACITY_IN_MIB;
+        } else {
+            return String.valueOf(StorageUtils.convertTo(size, "Mi"));
+        }
     }
 
-    protected static DiskCapacity of(String size) {
-        return new DiskCapacity(size);
+    /**
+     * Generate JBOD disk capacity configuration for a broker using the supplied storage configuration.
+     *
+     * @param storage Storage configuration for Kafka cluster.
+     * @param brokerId Id of the broker.
+     *
+     * @return Disk capacity configuration value for broker brokerId.
+     */
+    private Map<String, String> generateJbodDiskConfig(Storage storage, int brokerId) {
+        String size = "";
+        Map<String, String> diskConfig = new HashMap<>();
+
+        for (SingleVolumeStorage volume : ((JbodStorage) storage).getVolumes()) {
+            String name = VolumeUtils.createVolumePrefix(volume.getId(), true);
+            String path = KAFKA_MOUNT_PATH + "/" + name + "/" + KAFKA_LOG_DIR + brokerId;
+
+            if (volume instanceof PersistentClaimStorage) {
+                size = ((PersistentClaimStorage) volume).getSize();
+            } else if (volume instanceof EphemeralStorage) {
+                size = ((EphemeralStorage) volume).getSizeLimit();
+            }
```

Comment:
Maybe using the pattern matching for the above instanceOf as you did in the previous method?